### PR TITLE
[choreolib] Add AutoRoutine.observe()

### PR DESCRIPTION
**Pull request by BR88C**
Adds `AutoRoutine.observe(BooleanSupplier condition)`, which returns a `Trigger` bound to the routine's event loop. Naming is flexible.

### CODE REVIEW

**Comment by oh-yes-0-fps** [comment]
I'm still really not a big fan of the name, `.listen` makes me feel like I'm telling some action to listen for an event to happen and respond where this is really just sanitizing the scope of a `Trigger`. `.sanitize`, `.associate`, or maybe even like `.observe` if wanna stick with the implications that the `AutoRoutine` is doing something.

**Comment by BR88C** [comment]
I like `observe()`, implementing now

File: choreolib/src/main/java/choreo/auto/AutoRoutine.java [hunk 1]
```diff
@@ -107,6 +107,16 @@ public EventLoop loop() {
     return loop;
   }
 
+  /**
+   * Creates a {@link Trigger} that is bound to the routine's event loop.
+   *
+   * @param condition The condition represented by the trigger.
+   * @return A {@link Trigger} that is {@code true} when the condition is satisfied.
```
Comment (oh-yes-0-fps) [inline choreolib/src/main/java/choreo/auto/AutoRoutine.java:114]:
```suggestion
   * @return A {@link Trigger} that mirrors the state of the provided {@code condition}
```

File: choreolib/src/main/java/choreo/auto/AutoRoutine.java [hunk 2]
```diff
@@ -107,6 +107,16 @@ public EventLoop loop() {
     return loop;
   }
 
+  /**
+   * Creates a {@link Trigger} that is bound to the routine's event loop.
```
Comment (oh-yes-0-fps) [inline choreolib/src/main/java/choreo/auto/AutoRoutine.java:111]:
```suggestion
   * Creates a {@link Trigger} that is bound to the routine's {@link EventLoop}.
```

File: choreolib/src/main/java/choreo/auto/AutoRoutine.java (hunk 1 above)
Comment (oh-yes-0-fps) [reply choreolib/src/main/java/choreo/auto/AutoRoutine.java:114]: If you think this is worse just resolve the conversation

**Review by shueja — APPROVED** [review]

**Pull request merged**